lnwire/funding_locked.go

```diff
@@ -76,6 +90,13 @@ func (c *FundingLocked) Decode(r io.Reader, pver uint32) error {
 	if val, ok := typeMap[AliasScidRecordType]; ok && val == nil {
 		c.AliasScid = &aliasScid
 	}
+	if val, ok := typeMap[NonceRecordType]; ok && val == nil {
+		c.LocalNonce = &localNonce
+	}
+
+	if len(tlvRecords) != 0 {
```

Comment:
See comment above, this ensures that if we don't understand certain types then we still end up writing them to disk for future broadcast. 

IIUC, we also need to do this as otherwise the quick check tests start failing (raw bytes don't match the struct and other way around). 

Just to sync up on mental models here: this sets the `ExtraData` to what it would look like right after the call to `Encode`. Otherwise, the struct that had `Encode` called on it would have `ExtraData` populated to raw TLV, while the struct that had `Decode` on it wouldn't. This means a strict equality check fails.